test/core/test_lwt_mvar.ml

```diff
@@ -29,6 +29,34 @@ let suite =
          Lwt.return (Lwt.state y = Lwt.Return 0)
       );
 
+    test "take_available (full)"
+      (fun () ->
+         let x = Lwt_mvar.create 0 in
+         let y = Lwt_mvar.take_available x in
+         Lwt.return (y = Some 0)
+      );
```

Comment:
There should be another test to check that after `take_available` has been called, the mvar is empty. Something along the lines of:

```
test "take_available (twice)"
(fun () ->
  let m = Lwt_mvar.create 0 in
  let (_: int option) = Lwt_mvar.take_available m in
  let t = Let_mvar.take_available m in
  Lwt.return (t = None)
);
```